user/test/com/google/gwt/emultest/java/util/ObjectsTest.java

```diff
@@ -102,4 +102,69 @@ public void testHashCode() {
     Object obj = new Object();
     assertEquals(obj.hashCode(), Objects.hashCode(obj));
   }
+
+  public void testRequireNonNull() {
+    assertEquals("foo", Objects.requireNonNull(hideFromCompiler("foo")));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNull(hideFromCompiler(null)));
+  }
+
+  public void testRequireNonNullElse() {
+    assertEquals("foo", Objects.requireNonNullElse(hideFromCompiler("foo"), "bar"));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNullElse(hideFromCompiler(null), null));
+  }
+
+  public void testRequireNonNullElseGet() {
```

Comment:
Should also check that a non null default value is returned if null is provided.